### PR TITLE
Add support for Celo Blockchain

### PR DESCRIPTION
**Pull request by ravidsrk**
## Description

Add support for the [Celo](https://github.com/celo-org/celo-blockchain) blockchain . This PR supports sending funds.

Since Celo is fork of Geth, the ethereum signing logic is reused. The integration is very similar to the TomoChain and GoChain integration.

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: coins.json [hunk 1]
```diff
@@ -1404,5 +1404,26 @@
...
+        "publicKeyType": "secp256k1Extended",
+        "explorer": {
+            "url": "https://explorer.celo.org/",
+            "txPath": "/tx/",
+            "accountPath": "/address/"
+        },
+        "info": {
+            "url": "https://celo.org/",
+            "client": "https://github.com/celo-org/celo-blockchain",
+            "clientPublic": "https://rc1-forno.celo-testnet.org",
```
Comment (ravidsrk) [inline coins.json:1425]: RPC URL has testnet, but MainNet RPC as per this URL: https://docs.celo.org/developer-guide/overview/introduction/forno

File: docs/coins.md [hunk 2]
```diff
@@ -62,5 +62,6 @@ This list is generated from [./coins.json](../coins.json)
 | 6060    | GoChain          | GO     | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/gochain/info/logo.png" width="32" />      | <https://gochain.io>          |
 | 8964    | NULS             | NULS   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/nuls/info/logo.png" width="32" />         | <https://nuls.io>             |
 | 19167   | Zelcash          | ZEL    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/zelcash/info/logo.png" width="32" />      | <https://zel.cash>            |
+| 52752   | Celo             | CELO   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/celo/info/logo.png" width="32" />         | <https://celo.org/>           |
```
Comment (catenocrypt) [inline docs/coins.md:65]: This logo URL is not present in the assets repository!

File: swift/Podfile.lock [hunk 3]
```diff
@@ -17,4 +17,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 4382612b240ed6d49e9a3b9bbb107c581e72b8ea
 
-COCOAPODS: 1.9.3
+COCOAPODS: 1.9.1
```
Comment (catenocrypt) [inline swift/Podfile.lock:20]: Can you revert this file please?  Probably unintended.

Comment (ravidsrk) [reply swift/Podfile.lock]: @catenocrypt Fixed this

File: docs/coins.md (hunk 2 above)
Comment (ravidsrk) [reply docs/coins.md:65]: Added here: https://github.com/trustwallet/assets/pull/2485

File: coins.json (hunk 1 above)
Comment (hewigovens) [reply coins.json:1425]: I'm confused, is it mainnet or not?

Comment (ravidsrk) [reply coins.json:1425]: It's MainNet @hewigovens

**Comment by ravidsrk** [comment]
@hewigovens Celo is fork of ETH like Tomochain or Gochain. Do i still need to create a testcase for Transaction signing? 

It would be exactly like this: https://github.com/trustwallet/wallet-core/blob/master/android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/ethereum/TestEthereumTransactionSigner.kt

**Comment by hewigovens** [comment]
Tomochain and Gochain were added early, if you look at the other later added bitcoin forks, a mainnet tx + block explorer are mandatory (this will save us time to verify it's really working)

**Comment by catenocrypt** [comment]
We are still waiting for update on tests.

> Tomochain and Gochain were added early, if you look at the other later added bitcoin forks, a mainnet tx + block explorer are mandatory (this will save us time to verify it's really working)

**Comment by ravidsrk** [comment]
@catenocrypt I will add fixes this week.

**Pull request closed without merging**